### PR TITLE
how do I add the new task to the existing tasks?

### PR DESCRIPTION
**Pull request by linazoo**
how would you go about doing this?

### CODE REVIEW

File: src/components/InputDetail.tsx [hunk 1]
```diff
@@ -5,23 +5,19 @@ import Tab from "@mui/material/Tab";
...
   let theme = useTheme();
 
   const handleChange = (event: React.SyntheticEvent, selectedIndex: number) => {
     setValue(selectedIndex);
     const frequency = frequencies[selectedIndex];
-    console.log({ frequency });
+    // console.log({ frequency });
   };
 
   return (
```
Comment (johnlyden) [inline src/components/InputDetail.tsx:28]: you should have an `onChange` for each of these inputs.  You are passing in the functions to setTitle, setDescription, and setFrequency, so you'll want to use those in the `onChange`

for the title input, you should do something like:

```
onChange={(e) => {
    setTaskTitle(e.target.value)   
}}
```
and kinda do the same thing for the description input and the tabs onChange

File: src/components/InputDetail.tsx [hunk 2]
```diff
@@ -5,23 +5,19 @@ import Tab from "@mui/material/Tab";
...
 
-interface Props {
-  addTask: string;
-  setAddTask: React.Dispatch<React.SetStateAction<string>>;
-  handleAdd: (e: React.FormEvent) => void;
-}
 const frequencies = ["daily", "weekly", "monthly", "yearly"];
 
-const InputDetail: React.FC<Props> = ({ addTask, setAddTask, handleAdd }) => {
+const InputDetail: React.FC<AddTask> = ({ addTask, setAddTask, handleAdd }) => {
```
Comment (johnlyden) [inline src/components/InputDetail.tsx]: you'll need to destructure out the setter functions you passed in, so you can use them in the onchanges for each input

```suggestion
const InputDetail: React.FC<AddTask> = ({ setTaskTitle, setTaskDescription, setTaskFrequency }) => {
```

File: src/pages/AddPage.tsx [hunk 3]
```diff
@@ -5,10 +5,21 @@ import { AddTask } from "../components/model";
...
   const [addTask, setAddTask] = React.useState<string>("");
-  const [taskTitle, setTaskTitle] = React.useState<AddTask[]>([]);
+  const [addTasks, setAddTasks] = React.useState<AddTask[]>([]);
+  const [taskTitle, setTaskTitle] = React.useState<string>("");
+  const [taskDescription, setTaskDescription] = React.useState<string>("");
+  const [taskFrequency, setTaskFrequency] = React.useState<string>("");
 
   const handleAdd = (e: React.FormEvent) => {
     e.preventDefault();
+    // question: addpage > to add a new kitchen page
```
Comment (johnlyden) [inline src/pages/AddPage.tsx]: when `handleAdd` is called, `taskTitle` `taskDescription` and `taskFrequency` should have values.  you can use them here to do something like:

```
setAddTask({
title: taskTitle,
description: taskDescription,
frequency: taskFrequency
})
```

**Review by johnlyden — COMMENTED** [review]
looking good baby!  you're passing in the setter functions to `InputDetail` and that component should call those setters onChange of each input.

then when you click add, you can use those values to add that whole object to your list of tasks

File: src/pages/AddPage.tsx (hunk 3 above)
Comment (linazoo) [reply src/pages/AddPage.tsx]: Tried doing something like this but getting a type error: 

_Argument of type '{ title: string; description: string; frequency: string; }' is not assignable to parameter of type 'SetStateAction<string>'.
  Object literal may only specify known properties, and 'title' does not exist in type '(prevState: string) => string'.ts(2345)_

tried changing the type to an array but then it says it's not finding 'title' should I pass title, description and frequency into the addpage?

File: src/components/ListOfItems.tsx [hunk 4]
```diff
@@ -67,7 +68,13 @@ const ListOfItems = ({ items, onClick }: ListOfItemsProps) => {
               );
             })}
           </Stack>
-          <ColorButton variant="contained" sx={{ width: "100%", mt: "20px" }}>
+          <ColorButton
+            //@ts-ignore
```
Comment (johnlyden) [inline src/components/ListOfItems.tsx:72]: wtf???? why the ignore??? code smell much???

**Review by johnlyden — APPROVED** [review]
sad to see the ignore there but i'll allow it 🫢

**Pull request merged**